dom.bs

```diff
@@ -1186,33 +1187,70 @@ for discussion).
 
  <li><p>Initialize <var>event</var>'s {{Event/currentTarget}} attribute to <var>object</var>.
 
+ <li><p><a>Find</a> listener type matches for <var>object</var>.
+
  <li>
-  <p>For each <var>listener</var> in <var>listeners</var>, whose <b>removed flag</b> is unset, run
+  <p>If <var>event</var>'s <a>found listener flag</a> is unset, run
```

Comment:
"Find listener type matches for object" runs before the flag check. 

If there's an unprefixed event type match for a listener's type -> set the `found listener` flag (that way we don't bother to do legacy type check)
If there's no event type match for a listener's type, the flag is never set, so we can change the event type to the legacy one before running "Find listener type matches for object" again.

(I don't think there's any point to unsetting the flag at step 7, if that's what you're getting at.)